src/lando/main/models/revision.py

```diff
@@ -104,6 +115,89 @@ def serialize(self) -> dict[str, Any]:
+        parts = []
+        if self.author_name:
+            parts.append(self.author_name)
+        if self.author_email:
+            parts.append(f"<{self.author_email}>")
+
+        return " ".join(parts)
+
+    @property
+    def author_name(self) -> Optional[str]:
+        metadata = self._parse_metadata_from_patch()
+        return metadata.get("author_name")
+
+    @property
+    def author_email(self) -> Optional[str]:
+        metadata = self._parse_metadata_from_patch()
+        return metadata.get("author_email")
+
+    @property
+    def commit_message(self) -> Optional[str]:
+        metadata = self._parse_metadata_from_patch()
+        return metadata.get("commit_message")
+
+    @property
+    def timestamp(self) -> Optional[str]:
+        metadata = self._parse_metadata_from_patch()
+        if ts := metadata.get("timestamp"):
+            # Some codepaths (via Phabricator) have the timestamp set as an int.
+            # We make sure it's always a string.
+            return str(ts)
+        return None
+
+    def _parse_metadata_from_patch(self) -> dict[str, str]:
+        """Parse Hg metadata out of the raw patch, and update the patch_data if empty."""
+        if not self.patch_data:
+            commit_message = self.patch_helper.get_commit_description()
+            author_name, author_email = self._parse_author_string(
+                self.patch_helper.get_header("User")
+            )
+            timestamp = self.patch_helper.get_timestamp()
+
+            self.patch_data = {"commit_message": commit_message, "timestamp": timestamp}
+            if author_name:
+                self.patch_data["author_name"] = author_name
+            if author_email:
+                self.patch_data["author_email"] = author_email
+
+        return self.patch_data
+
+    @staticmethod
+    def _parse_author_string(author: str) -> tuple[str, str]:
+        """Parse a Git author string into author name and email.
+
+        The returned tuple will have the empty string "" for unmatched parts.
+        """
+        r = re.compile(
+            r"^(?P<author_name>.*?)? *<?(?P<author_email>[^ \t\n\r\f\v<]+@[^ \t\n\r\f\v>]+)>?"
+        )
+        m = r.match(author)
+        if not m:
+            return (author, "")
+        return m.groups()
+
+    @property
+    def diff(self) -> str:
+        """Return the unified diff text without any metadata"""
+        # The HgPatchHelper currently returns leading newline, which we don't want to
+        # return here, so we strip it.
+        return self.patch_helper.get_diff().lstrip()
+
+    @property
+    def patch_helper(self) -> HgPatchHelper:
+        """Create and cache an HgPatchHelper to parse the raw patch with Hg metadata."""
+        if not self._patch_helper:
+            self._patch_helper = HgPatchHelper(StringIO(self.patch))
+            if not self._patch_helper.diff_start_line:
+                raise NoDiffStartLine
+
+        return self._patch_helper
```

Comment:
These two methods are hg-specific, wonder if we should be making them more scm-agnostic?

Reply:
Yes, the purpose of this change is a stop gap to start hiding the HgPatchHelper.

Historically, we transformed Phab revision into a Hg Patch, which we no longer need to do.

In this first cut, we hide the patch helper from everything else, and let the interface of the Revision be sufficient, and we can change the implementation later on.